evalml/pipelines/components/transformers/encoders/onehot_encoder.py

```diff
@@ -111,7 +112,9 @@ def fit(self, X, y=None):
                 if top_n is None or len(value_counts) <= top_n:
                     unique_values = value_counts.index.tolist()
                 else:
-                    value_counts = value_counts.sample(frac=1, random_state=self.random_state)
+                    new_random_state = np.random.RandomState()
```

Comment:
Decided to define `new_random_state` here rather than at the top of the function because this is the only place it's needed. Happy to change it though!

Reply:
This looks fine to me!